ublox/src/ubx_packets/packets.rs

```diff
@@ -0,0 +1,718 @@
+use super::{
+    ubx_checksum, MemWriter, Position, UbxChecksumCalc, UbxPacketCreator, UbxPacketMeta,
+    UbxUnknownPacketRef, SYNC_CHAR_1, SYNC_CHAR_2,
+};
+use crate::error::{MemWriterError, ParserError};
+use bitflags::bitflags;
+use chrono::prelude::*;
+use ublox_derive::{
+    define_recv_packets, ubx_extend, ubx_extend_bitflags, ubx_packet_recv, ubx_packet_recv_send,
+    ubx_packet_send,
+};
+
+/// Geodetic Position Solution
+#[ubx_packet_recv]
+#[ubx(class = 1, id = 2, fixed_payload_len = 28)]
+struct NavPosLlh {
+    /// GPS Millisecond Time of Week
+    itow: u32,
+    /// Longitude
+    #[ubx(map_type = f64, scale = 1e-7, alias = lon_degrees)]
+    lon: i32,
+    /// Latitude
+    #[ubx(map_type = f64, scale = 1e-7, alias = lat_degrees)]
+    lat: i32,
+    /// Height above Ellipsoid
+    #[ubx(map_type = f64, scale = 1e-3)]
+    height_meters: i32,
+    /// Height above mean sea level
+    #[ubx(map_type = f64, scale = 1e-3)]
+    height_msl: i32,
+    /// Horizontal Accuracy Estimate
+    #[ubx(map_type = f64, scale = 1e-3)]
+    h_ack: u32,
+    /// Vertical Accuracy Estimate
+    #[ubx(map_type = f64, scale = 1e-3)]
+    v_acc: u32,
+}
+
+/// Velocity Solution in NED
+#[ubx_packet_recv]
+#[ubx(class = 1, id = 0x12, fixed_payload_len = 36)]
+struct NavVelNed {
+    /// GPS Millisecond Time of Week
+    itow: u32,
+    /// north velocity (m/s)
+    #[ubx(map_type = f64, scale = 1e-2)]
+    vel_north: i32,
+    /// east velocity (m/s)
+    #[ubx(map_type = f64, scale = 1e-2)]
+    vel_east: i32,
+    /// down velocity (m/s)
+    #[ubx(map_type = f64, scale = 1e-2)]
+    vel_down: i32,
+    /// Speed 3-D (m/s)
+    #[ubx(map_type = f64, scale = 1e-2)]
+    speed_3d: u32,
+    /// Ground speed (m/s)
+    #[ubx(map_type = f64, scale = 1e-2)]
+    ground_speed: u32,
+    /// Heading of motion 2-D (degrees)
+    #[ubx(map_type = f64, scale = 1e-5, alias = heading_degrees)]
+    heading: i32,
+    /// Speed Accuracy Estimate (m/s)
+    #[ubx(map_type = f64, scale = 1e-2)]
+    speed_accuracy_estimate: u32,
+    /// Course / Heading Accuracy Estimate (degrees)
+    #[ubx(map_type = f64, scale = 1e-5)]
+    course_heading_accuracy_estimate: u32,
+}
+
+/// Navigation Position Velocity Time Solution
+#[ubx_packet_recv]
+#[ubx(class = 1, id = 0x07, fixed_payload_len = 92)]
+struct NavPosVelTime {
+    /// GPS Millisecond Time of Week
+    itow: u32,
+    year: u16,
+    month: u8,
+    day: u8,
+    hour: u8,
+    min: u8,
+    sec: u8,
+    valid: u8,
+    time_accuracy: u32,
+    nanosecond: i32,
+    /// GNSS fix Type
+    #[ubx(map_type = GpsFix)]
+    fix_type: u8,
+    flags: u8,
+    flags2: u8,
+    num_satellites: u8,
+    #[ubx(map_type = f64, scale = 1e-7, alias = lon_degrees)]
+    lon: i32,
```

Comment:
Because of this field doesn't contain degrees. It contains degree multiply on 1e7.
If we add (in the future) interface to get "raw" fields, it would be good to not deceive user,
and distinguish between degrees/f64 and raw field where user need handle by him/her self.

The plan actually have the same units as getter everywhere if possible:
meters, meters per second, degrees. But while meters and seconds are obvious choice from [International System of Units](https://en.wikipedia.org/wiki/International_System_of_Units). 

Usage of degrees not so obvious, because of all functions from stdlib that work with angles works with radians in C/Rust and other standard libraries (like cos/sin/tan and so on),
so for degree I add such "comment" via method name, may be better return something like `struct Degree<T>(pub T)`.